src/igs_network.c

```diff
@@ -4109,7 +4115,14 @@ void igsagent_set_name (igsagent_t *agent, const char *name)
                                                 + strlen ("-IGS-SERVICES") + 1);
     snprintf (agent->services_channel, IGS_MAX_AGENT_NAME_LENGTH + 15,
               "%s-IGS-SERVICES", agent->definition->name);
-
+    
+    if (agent->igs_channel)
+        free (agent->igs_channel);
+    agent->igs_channel = (char *) zmalloc (strlen (agent->definition->name)
```

Comment:
With what has been said on the other comments, if it turns out that this channel is only to exchange SPLIT messages, I suggest renaming it to  
`snprintf (agent->igs_channel, IGS_MAX_AGENT_NAME_LENGTH + strlen("-IGS-SPLITS") + 1, "%s-IGS-SPLITS", agent->definition->name);`